### ISSUE TITLE
[@rancher/shell] Unable to load extensions on first run

### ISSUE THREAD

**Issue by rak-phillip** [rancher/dashboard#7424]
I'm unable to load custom extensions when first running a project with `@rancher/shell` as a dependency. Reloading the page, or staying logged in for subsequent runs will load the plugins. This behavior can be reproduced in the rancher plugins example repo with minimal modification to the project.  

## Steps to Reproduce

1. Clone the Rancher Examples repo `$ git clone git@github.com:rancher/ui-plugin-examples.git`
2. Install dependencies `yarn` (note: you will have to `$ rm yarn.lock` because of references to private registry)
3. Run the project `$ API=https://137.184.10.199 yarn dev`
4. If the plugins load, make sure to 
	1. Log out
	2. Close any active browsers tabs that are running Rancher Shell
	3. Stop running the project
	4. Run the project again `$ API=https://137.184.10.199 yarn dev`
	5. Log in to the dashboard

The plugins will not load. Refresh the page and the plugins will load.

## More Info

Tested with both `0.1.4` and `0.2.3` 
Tested in both Chrome and Firefox
Subsequent reloads will load the plugins
Staying logged in will load the plugins

**Comment by richard-cox** [comment]
This line interested me `Staying logged in will load the plugins`. I made a change in a soon to merge PR to `rancher/dashboard` `epinio-dev` that has a [bug fix in for logging out ](https://github.com/rancher/dashboard/compare/epinio-dev...richard-cox:dashboard:dex-pr#diff-1898a7b94117e7be24d1cb0e9c46083b62be96b947436ad725d71efa650ed01d), maybe logout is being called when there's no valid session... which is unloading the `pkg` plugins?

Possibly this part is calling logout on unauthed requests to Rancher
```
        // Go to the logout page for 401s, unless redirectUnauthorized specifically disables (for the login page)
        if ( opt.redirectUnauthorized !== false && res.status === 401 ) {
          dispatch('auth/logout', opt.logoutOnError, { root: true });
        }
```

**Comment by rak-phillip** [comment]
@richard-cox you are correct. The dashboard invokes logout when there's no valid session. This behavior is related to the the authenticated middleware, which will run through a routine the first time is is loaded; this includes dispatching an action to `auth/getUser` when auth is enabled and the user is not logged in.

https://github.com/rancher/dashboard/blob/afb24b8cc903d2659c4cdad9b6e3f1eac3017eee/shell/middleware/authenticated.js#L193

Ultimately, the store for Steve will dispatch `auth/logout` when attempting a request to `/v3/users?me=true`.

https://github.com/rancher/dashboard/blob/afb24b8cc903d2659c4cdad9b6e3f1eac3017eee/shell/plugins/steve/actions.js#L181-L183

After logging in, we don't attempt to init plugins again until the page refreshes. In addition, this issue only arises on the "first load" because this tied to the loading of the authenticated middleware. For example, we can reload the page before logging in and the plugins will load because the authenticated middleware has already loaded.

**Comment by rak-phillip** [comment]
In order to resolve this issue, we either need to 

1. init plugins again logging in
2. prevent plugins that are initialized from being removed during logout

**Comment by rak-phillip** [comment]
I see that [7b2288c8-a505-4316-af25-cbd1b6a8816a](https://github.com/rancher/dashboard/compare/epinio-dev...richard-cox:dashboard:dex-pr#diff-1898a7b94117e7be24d1cb0e9c46083b62be96b947436ad725d71efa650ed01d) has a fix that will prevent `builtin` plugins from being removed during logout. 

Testing this change looks to resolve the issue as it is described here.
